pkg/build/graph/analysis/bc.go

```diff
@@ -162,6 +162,62 @@ func FindCircularBuilds(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 	return markers
 }
 
+// findPendingTags is the guts behind FindPendingTags .... break out some of the content and reduce some indentation
+func findPendingTags(uncastIstNode graph.Node, g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+	istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+	if istNode.Found() {
+		return markers
+	}
+	bcNodes := buildedges.BuildConfigsForTag(g, uncastIstNode)
+	for _, bcNode := range bcNodes {
+		latestBuild := buildedges.GetLatestBuild(g, bcNode)
+
+		// A build config points to the non existent tag but no current build exists.
+		if latestBuild == nil {
+			markers = append(markers, osgraph.Marker{
+				Node:         graph.Node(bcNode),
+				RelatedNodes: []graph.Node{uncastIstNode},
+
+				Severity:   osgraph.WarningSeverity,
+				Key:        TagNotAvailableWarning,
+				Message:    fmt.Sprintf("%s needs to be imported or created by a build.", f.ResourceName(istNode)),
+				Suggestion: osgraph.Suggestion(fmt.Sprintf("oc start-build %s", f.ResourceName(bcNode))),
+			})
+			continue
+		}
+
+		// A build config points to the non existent tag but something is going on with
+		// the latest build.
+		// TODO: Handle other build phases.
+		switch latestBuild.Build.Status.Phase {
+		case buildapi.BuildPhaseCancelled:
+			// TODO: Add a warning here.
+		case buildapi.BuildPhaseError:
+			// TODO: Add a warning here.
+		case buildapi.BuildPhaseComplete:
+			// We should never hit this. The output of our build is missing but the build is complete.
+			// Most probably the user has messed up?
+		case buildapi.BuildPhaseFailed:
+			// Since the tag hasn't been populated yet, we assume there hasn't been a successful
+			// build so far.
+			markers = append(markers, osgraph.Marker{
+				Node:         graph.Node(latestBuild),
+				RelatedNodes: []graph.Node{uncastIstNode, graph.Node(bcNode)},
+
+				Severity:   osgraph.ErrorSeverity,
+				Key:        LatestBuildFailedErr,
+				Message:    fmt.Sprintf("%s has failed.", f.ResourceName(latestBuild)),
+				Suggestion: osgraph.Suggestion(fmt.Sprintf("Inspect the build failure with 'oc logs %s'", f.ResourceName(latestBuild))),
+			})
+		default:
+			// Do nothing when latest build is new, pending, or running.
```

Comment:
the change to make sure only 1 marker occurs for that (where multiple `oc start-build` can be listed) handles that perumtation now as well.